src/java/org/apache/cassandra/repair/PreviewRepairTask.java

```diff
@@ -128,7 +129,7 @@ private void maybeSnapshotReplicas(TimeUUID parentSession, String keyspace, List
             for (String table : mismatchingTables)
             {
                 // we can just check snapshot existence locally since the repair coordinator is always a replica (unlike in the read case)
-                if (!Keyspace.open(keyspace).getColumnFamilyStore(table).snapshotExists(snapshotName))
+                if (SnapshotManager.instance.getSnapshot(keyspace, table, snapshotName).isEmpty())
```

Comment:
I see a lot of calls checking for existence of a snapshot in the form of getSnapshot -> isEmpty. I wonder if we should add a dedicated method to check for snapshot existence?

Reply:
added two `exists` methods, one taking a predicate, another just ks name, cf name and snapshot name